### ISSUE TITLE
Node architecture does not match MSVC architecture

### ISSUE THREAD

**Issue by xhcoding** [cmake-js/cmake-js#189]
Whether --arch is X86 or X64, cl.exe is always x86, which leads to link error.

``` sh
F:\Code\Node\Echo>cmake-js --arch=x64 configure
[ 'C:\\Program Files\\nodejs\\node.exe',
  'C:\\Users\\user\\AppData\\Roaming\\npm\\node_modules\\cmake-js\\bin\\cmake-js',
  '--arch=x64',
  'configure' ]
info TOOL Using Visual Studio 15 2017 generator.
info CMD CONFIGURE
info RUN cmake "F:\Code\Node\Echo" --no-warn-unused-cli -G"Visual Studio 15 2017" -DCMAKE_JS_VERSION="5.3.1" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_RUNTIME_OUTPUT_DIRECTORY="F:\Code\Node\Echo\build" -DCMAKE_JS_INC="C:\Users\user\.cmake-js\node-x64\v10.16.1\include\node;F:\Code\Node\Echo\node_modules\nan" -DCMAKE_JS_SRC="C:/Users/user/AppData/Roaming/npm/node_modules/cmake-js/lib/cpp/win_delay_load_hook.cc" -DNODE_RUNTIME="node" -DNODE_RUNTIMEVERSION="10.16.1" -DNODE_ARCH="x64" -DCMAKE_JS_LIB="C:\Users\user\.cmake-js\node-x64\v10.16.1\win-x64\node.lib" -DCMAKE_SHARED_LINKER_FLAGS="/DELAYLOAD:NODE.EXE"
Not searching for unused variables given on the command line.
-- The C compiler identification is MSVC 19.16.27032.1
-- The CXX compiler identification is MSVC 19.16.27032.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: F:/Code/Node/Echo/build
```

**Comment by unbornchikken** [comment]
I've made some blind changes on master (I'm on macOS at work atm). Could you please test it?

**Comment by xhcoding** [comment]
It have a problem
···
C:\Users\user\AppData\Roaming\npm\node_modules\cmake-js\lib\es6\toolset.js:206
        if (ver === vswhereVersion || yield vsDetect.isInstalled(ver + ".0")) {
                                                    ^^^^^

SyntaxError: Unexpected strict mode reserved word
···
If I remove yield，it work well。

**Comment by unbornchikken** [comment]
Yeah, I've screwed up. Please test the latest.

**Comment by gjasny** [comment]
With the removed yield keyword cmake-js selects VS2019 on a system where only VS2017 is installed.

CMake generators:
```
  Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
* Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 14 2015 [arch] = Generates Visual Studio 2015 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 12 2013 [arch] = Generates Visual Studio 2013 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 11 2012 [arch] = Generates Visual Studio 2012 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 10 2010 [arch] = Generates Visual Studio 2010 project files.
                                 Optional [arch] can be "Win64" or "IA64".
  Visual Studio 9 2008 [arch]  = Generates Visual Studio 2008 project files.
                                 Optional [arch] can be "Win64" or "IA64".
```

vswhere:
```
vswhere -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationVersion
15.9.28307.770
```

**Comment by gjasny** [comment]
Looks like the yield keyword is needed :)

My clumsy change did the trick:
```diff
@@ -203,7 +204,12 @@ Toolset.prototype._getTopSupportedVisualStudioGenerator = async(function*() {
             }
         }

-        if (ver === vswhereVersion || yield vsDetect.isInstalled(ver + ".0")) {
+        if (yield vsDetect.isInstalled(ver + ".0")) {
+            result = gen;
+            maxVer = ver;
+        }
+
+        if (ver === vswhereVersion) {
             result = gen;
             maxVer = ver;
         }
```

**Comment by gjasny** [comment]
I'm wondering, for VS2019 how do you differentiate between `Win32` and `x64` without using the `-A` option?

**Comment by xhcoding** [comment]
> Looks like the yield keyword is needed :)
> 
> My clumsy change did the trick:
> 
> ```diff
> @@ -203,7 +204,12 @@ Toolset.prototype._getTopSupportedVisualStudioGenerator = async(function*() {
>              }
>          }
> 
> -        if (ver === vswhereVersion || yield vsDetect.isInstalled(ver + ".0")) {
> +        if (yield vsDetect.isInstalled(ver + ".0")) {
> +            result = gen;
> +            maxVer = ver;
> +        }
> +
> +        if (ver === vswhereVersion) {
>              result = gen;
>              maxVer = ver;
>          }
> ```

It's work for me. Thanks a lot. Hope merge to master.

**Comment by unbornchikken** [comment]
Please test master.

**Comment by unbornchikken** [comment]
Fixed as of 5.2.2.

**Comment by unbornchikken** [comment]
I mean 5.3.2
